### PR TITLE
bump snowflake driver version

### PR DESCRIPTION
**Pull request by dpsutton**
bump snowflake

### CODE REVIEW

File: modules/drivers/snowflake/deps.edn [hunk 1]
```diff
@@ -2,4 +2,4 @@
  ["src" "resources"]
 
  :deps
- {net.snowflake/snowflake-jdbc {:mvn/version "3.15.1"}}}
+ {net.snowflake/snowflake-jdbc {:mvn/version "3.17.0"}}}
```
Comment (camsaul) [inline modules/drivers/snowflake/deps.edn:5]: [They say 3.19 is out now](https://docs.snowflake.com/en/release-notes/clients-drivers/jdbc-2024), any reason not to bump to latest?

```suggestion
 {net.snowflake/snowflake-jdbc {:mvn/version "3.19.0"}}}
```

**Review by camsaul — APPROVED** [review]
Omg finally lol. LFGGGGG

**Comment by paoliniluis** [comment]
You can't even add the db...
image

**Review by dragonsahead — CHANGES_REQUESTED** [review]
I know this is just a driver bump, but we did it to test java 17+ and snowflake driver, which doesn't work. We need to think about "--add-opens java.base/java.nio=ALL-UNNAMED" and evaluate the consequences
https://community.snowflake.com/s/article/JDBC-Driver-Compatibility-Issue-With-JDK-16-and-Later

UPDATE: approving since this is a driver bump

**Review by dragonsahead — APPROVED** [review]

**Comment by camsaul** [comment]
The changelog seems to suggest that 3.17.0+ supports Java 21 but I guess that might or might not be true

https://docs.snowflake.com/en/release-notes/clients-drivers/jdbc-2024

> Added support for Java version 21.

**Pull request merged**